ci/lint/Dockerfile

```diff
@@ -0,0 +1,29 @@
+# See test/lint/README.md for usage.
+#
+# This container basically has to live in this directory in order to pull in the CI
+# install scripts. If it lived in the root directory, it would have to pull in the
+# entire repo as docker context during build; if it lived elsewhere, it wouldn't be
```

Comment:
> If it lived in the root directory, it would have to pull in the entire repo as docker context during build;

Would you have preferred it lived in root? I think it makes more sense in `/ci/lint`, but if it would be better you could use `.dockerignore` to prevent adding everything to your docker context, as in this diff:

<details>
<summary>git diff</summary>

```diff
diff --git a/.dockerignore b/.dockerignore
new file mode 100644
index 000000000..142df6af9
--- /dev/null
+++ b/.dockerignore
@@ -0,0 +1,2 @@
+*
+!/ci/lint
\ No newline at end of file
diff --git a/ci/lint/Dockerfile b/Dockerfile
similarity index 91%
rename from ci/lint/Dockerfile
rename to Dockerfile
index 03c20c728..87c9bb85a 100644
--- a/ci/lint/Dockerfile
+++ b/Dockerfile
@@ -16,8 +16,8 @@ ENV LC_ALL=C.UTF-8
 ENV SKIP_PYTHON_INSTALL=1
 
 # Must be built from ./ci/lint/ for these paths to work.
-COPY ./docker-entrypoint.sh /entrypoint.sh
-COPY ./04_install.sh /install.sh
+COPY ./ci/lint/docker-entrypoint.sh /entrypoint.sh
+COPY ./ci/lint/04_install.sh /install.sh
 
 RUN /install.sh && \
   echo 'alias lint="./ci/lint/06_script.sh"' >> ~/.bashrc && \
```
</details>